callback_plugins/help.py

```diff
@@ -48,29 +47,67 @@ def __init__(self):
         self.handler = logging.FileHandler(FAIL_LOGFILE)
         self.logger.addHandler(self.handler)
 
-    def format_results(self, result):
-        messages = '[' + str(result._host.name) + ']: Ansible Failed! => changed=' + \
-                   str(result._result['changed']) + '\n'
-        for k, v in result._result.iteritems():
-            if isinstance(v, list):
-                messages += '  ' + str(k) + ':' + '\n'
-                for key, value in v[0].iteritems():
-                    if 'ansible' not in key and key != 'invocation':
-                        messages += '    - ' + str(key) + ': ' + str(value) + '\n'
-            elif k != 'changed' and 'ansible' not in k and k != 'invocation' and k != 'exception':
-                messages += '  ' + str(k) + ': ' + str(v) + '\n'
-        return messages
+    def _format_results(self, result, indent=None, sort_keys=True, keep_invocation=False):
+        # All result keys stating with _ansible_ are internal, so remove them from the result before we output anything.
+        abridged_result = strip_internal_keys(result._result)
+
+        # remove invocation unless specifically wanting it
+        if not keep_invocation and self._display.verbosity < 3 and 'invocation' in abridged_result:
+            del abridged_result['invocation']
+
+        # remove diff information from screen output
+        if self._display.verbosity < 3 and 'diff' in abridged_result:
+            del abridged_result['diff']
+
+        if 'access_control_allow_headers' in abridged_result:
```

Comment:
`dict.pop()` is more elegant , see https://stackoverflow.com/questions/11277432/how-to-remove-a-key-from-a-python-dictionary